test/unit/block_test.rb

```diff
@@ -31,6 +31,10 @@ def test_op_write_raw_w
     assert_equal(source, template.render!)
   end
 
+  def test_invalid_blocks
+    assert_raises(Liquid::SyntaxError) { Liquid::Template.parse("{%-%}") }
```

Comment:
We should have a similar test for an empty variable with a single strip character (`Liquid::Template.parse("{{-}}")`) which I don't think results in a syntax error.  That also brings implications of how stripping should be handled, since this seems like an edge case that could result in stripping only one side or both sides.